components/brave_federated/brave_federated_service.cc

```diff
@@ -42,49 +43,41 @@ void BraveFederatedService::RegisterProfilePrefs(PrefRegistrySimple* registry) {
   OperationalPatterns::RegisterPrefs(registry);
 }
 
-void BraveFederatedService::InitPrefChangeRegistrar() {
-  local_state_change_registrar_.Init(local_state_);
-  local_state_change_registrar_.Add(
-      brave::kP3AEnabled,
-      base::BindRepeating(&BraveFederatedService::OnPreferenceChanged,
-                          base::Unretained(this)));
-}
+void BraveFederatedService::Init() {
+  VLOG(1) << "Initialising federated service";
 
-void BraveFederatedService::Start() {
   base::FilePath db_path(
       browser_context_path_.AppendASCII("data_store.sqlite"));
-
   data_store_service_.reset(new DataStoreService(db_path));
   data_store_service_->Init();
 
   eligibility_service_.reset(new EligibilityService());
 
-  if (ShouldStartOperationalPatterns()) {
-    operational_patterns_.reset(
-        new OperationalPatterns(prefs_, url_loader_factory_));
-    operational_patterns_->Start();
-  }
+  operational_patterns_.reset(
+      new OperationalPatterns(prefs_, url_loader_factory_));
+  MaybeStartOperationalPatterns();
+  MaybeStopOperationalPatterns();
 }
 
 DataStoreService* BraveFederatedService::GetDataStoreService() const {
   DCHECK(data_store_service_);
   return data_store_service_.get();
 }
 
-bool BraveFederatedService::ShouldStartOperationalPatterns() {
-  return IsP3AEnabled() && IsOperationalPatternsEnabled();
+///////////////////////////////////////////////////////////////////////////////
+
+void BraveFederatedService::InitPrefChangeRegistrar() {
+  local_state_change_registrar_.Init(local_state_);
+  local_state_change_registrar_.Add(
+      brave::kP3AEnabled,
+      base::BindRepeating(&BraveFederatedService::OnPreferenceChanged,
+                          base::Unretained(this)));
 }
 
 void BraveFederatedService::OnPreferenceChanged(const std::string& key) {
-  if (operational_patterns_) {
-    if (ShouldStartOperationalPatterns()) {
-      operational_patterns_->Start();
-    } else {
-      operational_patterns_->Stop();
-    }
-  } else {
-    Start();
-  }
+  DCHECK(operational_patterns_);
```

Comment:
should be moved to `MaybeStartOperationalPatterns` and `MaybeStopOperationalPatterns`